sdk/core/core-client/src/serializationPolicy.ts

```diff
@@ -91,6 +94,16 @@ function serializeHeaders(
       }
     }
   }
+  if (operationArguments.options?.requestOptions?.customHeaders) {
```

Comment:
nit: you could create a custom headers variable before the if and use it afterwards to save some characters

```typescript
const customHeaders = operationArguments.options?.requestOptions?.customHeaders;
if (customHeaders )
```